### PR TITLE
Feature/memoize optimisations

### PR DESCRIPTION
**Pull request by noway**

### CODE REVIEW

**Comment by noway** [comment]
Went from this:
Screen Shot 2021-09-11 at 4 56 43 PM
To this:
Screen Shot 2021-09-11 at 5 51 16 PM

59.8ms -> 2.1ms (28x improvement) of render method when clicking a day on the calendar

File: src/calendar/Calendar.tsx [hunk 1]
```diff
@@ -54,81 +56,110 @@ export const BookingCalendar: FunctionComponent<BookingCalendarProps> = ({
...
+              locations={locations}
+            />
+          );
+        })}
+      </MonthContainer>
+    </CalendarSectionContainer>
+  );
+}
+// FOR FUTURE: ideally we just have cheaper calendar month containers? Idk. Maybe useMemo or memoizeOne is a more elegant solution? Idk. This will do for now.
+const CalendarMonthContainer = React.memo(CalendarMonthContainerExpensive);
```
Comment (MinhPhu0304) [inline src/calendar/Calendar.tsx:116]: Good stuff

**Review by oeed — APPROVED** [review]
Definitely more snappier. Implementation looks good to me.

**Review by MinhPhu0304 — APPROVED** [review]

**Pull request merged**